examples/run_mock_client_main.cc

```diff
@@ -2,52 +2,136 @@
+
+template <typename T, typename U>
+void LogRequestAndResponse(T request, StatusOr<U> reply_or) {
+  LOG(INFO) << "Request sent: \n" << request.DebugString();
+  if (reply_or.ok()) {
+    LOG(INFO) << "Response received: \n" << reply_or.ValueOrDie().DebugString();
+  } else {
+    LOG(ERROR) << "Request failed: \n" << reply_or.status().ToString();
+  }
+}
 
 int main(int argc, char** argv) {
+  gfs::common::SystemLogger::GetInstance().Initialize(/*program_name=*/argv[0]);
+
   // Initialize an instance of communication manager
   // TODO(tugan): add support to listen on host:port based on configuration
-  std::string server_address("0.0.0.0:50051");
+  std::string master_address("0.0.0.0:50051");
+  std::string chunk_server_address("0.0.0.0:50052");
   auto credentials = grpc::InsecureChannelCredentials();
-  auto master_channel = grpc::CreateChannel(server_address, credentials);
+  auto master_channel = grpc::CreateChannel(master_address, credentials);
+  auto chunk_server_lease_channel =
+      grpc::CreateChannel(chunk_server_address, credentials);
   MasterMetadataServiceClient metadata_client(master_channel);
+  ChunkServerServiceClient chunk_server_client(chunk_server_lease_channel);
 
-  // Prepare a mock gRPC request and client context
+  // Prepare a mock gRPC: OpenFile
   OpenFileRequest open_request;
-  grpc::ClientContext client_context1;
   open_request.set_filename("/tmp/test_open_file");
   open_request.set_chunk_index(0);
   open_request.set_mode(OpenFileRequest::WRITE);
   open_request.set_create_if_not_exists(true);
 
-  // Send the request and print the response to screen
-  StatusOr<OpenFileReply> status_or =
-      metadata_client.SendRequest(open_request, client_context1);
-  std::cout << "Request sent: \n" << open_request.DebugString() << std::endl;
-
-  if (status_or.ok()) {
-    std::cout << "Response received: \n"
-              << status_or.ValueOrDie().DebugString() << std::endl;
-  } else {
-    std::cout << "Request failed: \n"
-              << status_or.status().ToString() << std::endl;
-  }
+  grpc::ClientContext client_context;
+  LogRequestAndResponse(
+      open_request, metadata_client.SendRequest(open_request, client_context));
 
+  // Prepare a mock gRPC: DeleteFile
   DeleteFileRequest delete_request;
   delete_request.set_filename("/tmp/test_delete_file");
+
   grpc::ClientContext client_context2;
   Status status = metadata_client.SendRequest(delete_request, client_context2);
-  std::cout << "Request sent: \n" << delete_request.DebugString() << std::endl;
-  std::cout << "Request status: \n" << status.ToString() << std::endl;
+  LOG(INFO) << "Request sent: " << delete_request.DebugString();
+  LOG(INFO) << "Request status: " << status.ToString();
+
+  // Prepare a mock gRPC: GrantLease
+  google::protobuf::Timestamp lease_expiration_time;
+  lease_expiration_time.set_seconds(1000);
+  GrantLeaseRequest grant_lease_request;
+  grant_lease_request.set_chunk_handle("/tmp/test_grant_lease");
```

Comment:
I think "/tmp/test_grant_lease" is suggesting a file name instead of chunk_handle? To avoid confusion I think it is better to use a chunk_handle that is consistent with how it is generated, e.g. a 64-bit long integer.

Reply:
Used an actual UUID string instead.
However, do note that in protobuf definition, a chunk handle is a STRING , not an integer , by design, since we designed it under the assumption it is UUID string, not integer.

Also, the paper only suggests it's a 64-bit handle, not necessary an integer; a string is always more extensible in terms of system and scalability, especially when a system may evolve to support larger set of file handles, or change how a chunk handle is defined